Move CB benchmarking and accuracy from the 'samples' to 'tools' (#1026)
openvinotoolkit/openvino.genai: samples/CMakeLists.txt

            cpp/lora_greedy_causal_lm
            cpp/multinomial_causal_lm

>             # Don't install prompt_lookup_decoding_lm and speculative_decoding_lm because they don't use openvino_genai library and arent verifyed yet.

Reviewer: ```suggestion
            # Don't install prompt_lookup_decoding_lm because it doesn't use openvino_genai library and is not verified yet.
```
